### PR TITLE
[PF-218]: Creates storage logs

### PR DESCRIPTION
**Pull request by yonghaoy**
This has 180 days life cycle and an WRITER ACL. 
See: https://github.com/broadinstitute/gcp-dm-templates/blob/eeee90fa4619b273d07206a867b01914cdeb0a30/templates/project.py#L165

### CODE REVIEW

File: src/main/java/bio/terra/rbs/service/resource/flight/CreateBucketStep.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import bio.terra.stairway.StepResult;
+import com.google.cloud.storage.Acl;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.StorageOptions;
+import com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Creates the basic GCP project. */
+public class CreateBucketStep implements Step {
```
Comment (wchamber) [inline src/main/java/bio/terra/rbs/service/resource/flight/CreateBucketStep.java:20]: Fix the comment. Improve the class name. This is not a step to create a generic bucket, it's specifically for storage logs.
Make sure the javadoc description for this class includes why we want to do such a thing.

File: src/test/java/bio/terra/rbs/integration/CreateProjectFlightIntegrationTest.java [hunk 2]
```diff
@@ -382,6 +408,17 @@ private void assertIamBindingsContains(Project project, List<IamBinding> iamBind
     }
   }
 
+  private void assertBucketExists(Project project) throws Exception {
```
Comment (wchamber) [inline src/test/java/bio/terra/rbs/integration/CreateProjectFlightIntegrationTest.java:411]: assertLogStorageBucketExists?

**Review by wchamber — CHANGES_REQUESTED** [review]

File: src/main/java/bio/terra/rbs/service/resource/flight/CreateBucketStep.java (hunk 1 above)
Comment (yonghaoy) [reply src/main/java/bio/terra/rbs/service/resource/flight/CreateBucketStep.java:20]: Done. I was thinking let this steps handles all buckets creation. But we can have make each step creates one bucket if we have more..

File: src/main/java/bio/terra/rbs/service/resource/flight/CreateStorageLogBucketStep.java [hunk 3]
```diff
@@ -16,9 +15,13 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** Creates the basic GCP project. */
-public class CreateBucketStep implements Step {
-  private final Logger logger = LoggerFactory.getLogger(CreateBucketStep.class);
+/**
+ * Creates the Storage log bucket for the project.
+ *
+ * <p>See <a href="https://cloud.google.com/storage/docs/access-logs">Storage logs introduction</a>
```
Comment (wchamber) [inline src/main/java/bio/terra/rbs/service/resource/flight/CreateStorageLogBucketStep.java:21]: per the link, don't we have to do step 3 of enabling logging on the bucket? From my understanding of this link, this is how to allow access logs for the bucket to be written. But the comments on Deployment Manager are about logging GCE stuff to this bucket. What is this bucket really for?

Comment (yonghaoy) [reply src/main/java/bio/terra/rbs/service/resource/flight/CreateStorageLogBucketStep.java:21]: Just checked with Doug, the DM comment is copt/paste error from export bucket:
https://github.com/broadinstitute/gcp-dm-templates/blob/eeee90fa4619b273d07206a867b01914cdeb0a30/templates/project.py#L117
we enable logging on the workspace bucket when it is created, not this storage log bucket. The bucket here is where are logs are written. We do step 1 and 2 here, step 3 is in the workspace bucket. 

Changed the comment little bit

Comment (yonghaoy) [reply src/main/java/bio/terra/rbs/service/resource/flight/CreateStorageLogBucketStep.java]: added comment

File: src/main/java/bio/terra/rbs/service/pool/GcpResourceConfigValidator.java [hunk 4]
```diff
@@ -17,7 +17,8 @@
  */
 public class GcpResourceConfigValidator implements ResourceConfigValidator {
   /** List of services required to be enabled. */
-  private static final List<String> REQUIRED_SERVICES = ImmutableList.of("compute.googleapis.com");
+  private static final List<String> REQUIRED_SERVICES =
```
Comment (wchamber) [inline src/main/java/bio/terra/rbs/service/pool/GcpResourceConfigValidator.java:20]: not related to this PR - but if we update validation, we need to make sure old deactivated pools don't need to be modified. Just thought of that, I don't know if that's already the case.

**Review by wchamber — APPROVED** [review]

File: src/main/java/bio/terra/rbs/service/pool/GcpResourceConfigValidator.java (hunk 4 above)
Comment (yonghaoy) [reply src/main/java/bio/terra/rbs/service/pool/GcpResourceConfigValidator.java:20]: this validate configs loaded from file. And the file only has activate pool. Should be fine

**Pull request merged**